app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -92,4 +92,39 @@ def allowed_storages(options = {})
 
     result.select { |s| s.storage_domain_type == "data" }
   end
+
+  def load_allowed_vlans(hosts, vlans)
+    ems = source_ems
+    if update_vnic_profile_supported?(ems)
+      load_v4_allowed_vlans(ems, vlans)
+    else
+      super(hosts, vlans)
+    end
+  end
+
+  def source_ems
+    src = get_source_and_targets
+    load_ar_obj(src[:ems])
+  end
+
+  def update_vnic_profile_supported?(ems)
+    ems.supports_update_vnic_profile?
+  end
+
+  def load_v4_allowed_vlans(ems, vlans)
+    ems.with_provider_connection(:version => 4) do |connection|
```

Comment:
@Ladas yeah we had issues with this for dvswitches also, a lot of times the UI worker can't even route to the provider directly so it can fail to connect at all.

If we merge this as is we should create an issue to track needing to add this to vmdb so we don't forget about it